reports/cost_variation.go

```diff
@@ -0,0 +1,251 @@
+//   Copyright 2018 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package reports
+
+import (
+"context"
+"database/sql"
+"fmt"
+"strconv"
+"strings"
+"time"
+
+"github.com/360EntSecGroup-Skylar/excelize"
+"github.com/trackit/jsonlog"
+
+"github.com/trackit/trackit-server/aws"
+"github.com/trackit/trackit-server/aws/usageReports/history"
+"github.com/trackit/trackit-server/costs/diff"
+)
+
+type costVariationProduct map[time.Time]diff.PricePoint
+type costVariationReport map[string]costVariationProduct
+
+const (
+	freqDaily = iota
+	freqMonthly
+)
```

Comment:
Add a slice to transform freq* in string
`[]string{"day", "month"}`

Reply:
I created a `struct` to be able to support different frequency if needed later.